src/shadowbox/integration_test/test.sh

```diff
@@ -109,8 +132,8 @@ function cleanup() {
     fail "Client should not have access to target IP" || (($? == 28))
 
   # Exit code 6 for "Could not resolve host".
-  docker exec $CLIENT_CONTAINER curl --silent --connect-timeout 1 http://target > /dev/null && \
-    fail "Client should not have access to target host" || (($? == 6))
+  test_with_retries "docker exec $CLIENT_CONTAINER curl --silent --connect-timeout 1 http://target" \
```

Comment:
It's a bit harder to read this now. There are two failure conditions and test_with_retries is doing too much.

What do you think about creating a function that runs the previous test, then pass it to a retry function?
Example:
```bash
function client_does_not_have_access_to_target_host() {
  docker exec $CLIENT_CONTAINER curl --silent --connect-timeout 1 http://target > /dev/null && fail "Client should not have access to target host" || (($? == 6))
}
...
test_with_retries client_does_not_have_access_to_target_host
...
test_with_retries deleted_access_key_is_inactive
```

Reply:
That's a great suggestion. Done.